nautobot/dcim/views.py

```diff
@@ -1310,7 +1310,12 @@ def get_extra_context(self, request, instance):
             "_path__destination",
             "tags",
         )
-        interface_table = tables.DeviceInterfaceTable(data=interfaces, user=request.user, orderable=False)
+        interface_table = tables.DeviceInterfaceTable(
+            is_vc_master=VirtualChassis.objects.filter(master=instance).exists(),
+            data=interfaces,
+            user=request.user,
+            orderable=False,
+        )
         if request.user.has_perm("dcim.change_interface") or request.user.has_perm("dcim.delete_interface"):
             interface_table.columns.show("pk")
```

Comment:
Actually, could follow this pattern here instead of adding it as a parameter to `DeviceInterfaceTable.__init__()`:

```python
if VirtualChassis.objects.filter(master=instance).exists():
    interface_table.columns.show("device")
```

Reply:
Cool will test now.